protos/component_metadata/component_metadata.proto

```diff
@@ -0,0 +1,90 @@
+syntax = "proto3";
+
+package mavsdk.rpc.component_metadata;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.component_metadata";
+option java_outer_classname = "ComponentMetadataProto";
+
+// Access component metadata json definitions, such as parameters.
+service ComponentMetadataService {
+
+    /*
+     * Request metadata from a specific component. This is used to start requesting metadata from a component.
+     * The metadata can later be accessed via subscription (see below) or GetMetadata.
+     */
+    rpc RequestComponent(RequestComponentRequest) returns(RequestComponentResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Request metadata from the autopilot component. This is used to start requesting metadata from the autopilot.
+     * The metadata can later be accessed via subscription (see below) or GetMetadata.
+     */
+    rpc RequestAutopilotComponent(RequestAutopilotComponentRequest) returns(RequestAutopilotComponentResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Register a callback that gets called when metadata is available
+     */
+    rpc SubscribeMetadataAvailable(MetadataAvailableRequest) returns(stream MetadataUpdateResponse) { option (mavsdk.options.async_type) = ASYNC; }
+
+    /*
+     * Access metadata. This can be used if you know the metadata is available already, otherwise use
+     * the subscription to get notified when it becomes available.
+     */
+    rpc GetMetadata(GetMetadataRequest) returns(GetMetadataResponse) { option (mavsdk.options.async_type) = SYNC; }
+}
+
+message RequestComponentRequest {
+    uint32 compid = 1; // The component ID to request
+}
+
+message GetMetadataRequest {
+    uint32 compid = 1; // The component ID to request
+    MetadataType type = 2; // The metadata type
+}
+
+message GetMetadataResponse {
+    MetadataResult result = 1;
+    MetadataData response = 2;
+}
+
+// Metadata response
+message MetadataData {
+    string json_metadata = 1; // The JSON metadata
+}
+
+// Result type.
+message MetadataResult {
+    // Possible results returned for GetMetadata
+    enum Result {
+        RESULT_SUCCESS = 0; // Success
+        RESULT_NOT_AVAILABLE = 1; // Metadata is not available
```

Comment:
I would think you also need `CONNECTION_ERROR`, `UNSUPPORTED`, `DENIED`, `FAILED` in case the `request_message` call fails with a nack.

Reply:
I added them